pulp_rpm/app/models.py

```diff
@@ -1,33 +1,203 @@
+            First byte of the header
+        rpm_license (Text):
+            License term applicable to the package software (GPLv2, etc.)
+        rpm_packager (Text):
+            Person or persons responsible for creating the package
+        rpm_sourcerpm (Text):
+            Name of the source package (srpm) the package was built from
+        rpm_vendor (Text):
+            Name of the organization that produced the package
 
+        size_archive (BigInteger):
+            Size, in bytes, of the archive portion of the original package file
+        size_installed (BigInteger):
+            Total size, in bytes, of every file installed by this package
+        size_package (BigInteger):
+            Size, in bytes, of the package
+
+        time_build (BigInteger):
+             Time the package was built in seconds since the epoch.
+        time_file (BigInteger):
+            The mtime of the package file in seconds since the epoch; this is the 'file' time
+            attribute in the primary XML.
     """
-    TYPE = 'type'
+    TYPE = 'rpm'
+
+    # Required metadata
+    name = models.TextField()
+    epoch = models.TextField()
+    version = models.TextField()
+    release = models.TextField()
+    arch = models.TextField()
+
+    pkgId = models.TextField(unique=True)  # formerly "checksum"
+    checksum_type = models.TextField(choices=CHECKSUM_CHOICES)
+
+    # Optional metadata
+    summary = models.TextField(blank=True)
+    description = models.TextField(blank=True)
+    url = models.TextField(blank=True)
+
+    # A string containing a JSON-encoded list of dictionaries, each of which represents a single
+    # changelog. Each changelog dict contains the following fields:
+    #
+    #   date (int):     date of changelog - seconds since epoch
+    #   author (str):   author of the changelog
+    #   changelog (str: changelog text
+    changelogs = models.TextField(default='[]', blank=True)
 
-    # field1 = models.TextField(blank=False, null=False)
-    # field2 = models.TextField(blank=False, null=False)
+    # A string containing a JSON-encoded list of dictionaries, each of which represents a single
+    # file. Each file dict contains the following fields:
+    #
+    #   type (str):     one of "" (regular file), "dir", "ghost"
+    #   path (str):     path to file
+    #   name (str):     filename
+    files = models.TextField(default='[]', blank=True)
 
-    # class Meta:
-    #     unique_together = (
-    #         'field1',
-    #         'field2'
-    #     )
+    # Each of these is a string containing a JSON-encoded list of dictionaries, each of which
+    # represents a dependency. Each dependency dict contains the following fields:
+    #
+    #   name (str):     name
+    #   flags (str):    flags
+    #   epoch (str):    epoch
+    #   version (str):  version
+    #   release (str):  release
+    #   pre (bool):     preinstall
+    requires = models.TextField(default='[]', blank=True)
```

Comment:
How do 'rich dependencies' fit in here?  I couldn't find any information with respect to createrepo_c + 'rich' or 'boolean' dependencies.

Reply:
Yeah, I believe we should not be concerned about that for the modeling part.
Any dependency solving should be handled by an external library and it's its business to figure out all the relationships and types of dependencies the package has by repository and package metadata.